datumaro/components/visualizer.py

```diff
@@ -192,40 +195,98 @@ def _sort_key(ann: Annotation):
 
         return sorted(annotations, key=_sort_key)
 
+    def get_random_items(self, n_samples: int) -> List[DatasetItem]:
+        """Get random samples from the dataset"""
+        if n_samples >= len(self.dataset):
+            raise ValueError(
+                f"n_samples={n_samples} should be less than the dataset size ({len(self.dataset)})."
+            )
+
+        return random.choices(self._items, k=n_samples)
+
     def vis_gallery(
         self,
-        ids: List[Union[str, DatasetItem]],
-        subset: Optional[Union[str, List[str]]] = None,
+        ids: Optional[List[str]] = None,
+        subsets: Optional[Union[str, List[str]]] = None,
+        items: Optional[List[DatasetItem]] = None,
```

Comment:
The reason why you added the `items` as `input` is to consider we only put `items` in the `input`? I think users could consider need to set all of `ids, subsets, items` as `input`.

Reply:
It's true. I introduced `typing.overload`s and updated docstrings for them.